andela/ah-infinity-stones
# #161966916 Implement Article CRUD

### What does this PR do?
Implement the article CRUD

### Description of Task to be completed
Implement endpoints for
* Creating an article 
* Getting all articles 
* Retrieving a specific article by id
* Update a specific article
* Delete a specific article
* Adding slug functionality
* Configure the code to pep8 standards
* Refactor tests

### How should this be manually tested?
* `git clone -b ft-create-article-161966916 https://github.com/andela/ah-infinity-stones` and ` cd ah-infinity-stones`.
* Run `python3 -m venv virtual` to create a virtual environment and `source virtual/bin/activate` to activate it.
* `pip install -r requirements.txt` to install dependencies
* Create a test database in the postgres shell
* Set up environment variables using the sample below as a guide:
```export DATABASE_USER='<postgres username>'
export DATABASE_PASSWORD='<postgres user's password>'
export HOST='localhost'
export PORT='5432'
export DATABASE_NAME='<name_of_database>' 
export SECRET_KEY='<personal_secret_key>'
export EMAIL_HOST_USER='<email_host_username>'
export EMAIL_HOST_PASSWORD='<email_host_password>'
```
* Run `python3 manage.py makemigrations` followed by `python3 manage.py migrate` to apply migrations
 Acess `http://127.0.0.1:8000/api/users/` through [Postman](https://www.getpostman.com/), provide inputs in the following format and create account.
```
 {
	"user": {
		"username": "remmykn",
		"email": "remmy@test.com",
		"password": "Password@123"
	}
}

```
Check your email and activate the account. Use the registration token to access the following endpoints:
- `http://127.0.0.1:8000/api/articles` - Post an article and get all articles
-`http://127.0.0.1:8000/api/articles/<slug>` - To get a specific article to either update or delete.
*  Run tests`python3 manage.py test authors.apps.articles.tests.test_views`

### Relevant pivotal tracker stories
[Delivers #161966916](https://www.pivotaltracker.com/story/show/161966916)

### Screenshots
![image](https://user-images.githubusercontent.com/17976289/50316232-d4b24780-04c6-11e9-8562-6fe4304e3e94.png)

![image](https://user-images.githubusercontent.com/17976289/50316296-19d67980-04c7-11e9-8e52-20cc911783b3.png)

![image](https://user-images.githubusercontent.com/17976289/50316359-53a78000-04c7-11e9-88de-6d0757dbc09a.png)

![image](https://user-images.githubusercontent.com/17976289/50316420-95d0c180-04c7-11e9-9e97-a1f0259cf056.png)

![image](https://user-images.githubusercontent.com/17976289/50316432-a3864700-04c7-11e9-8014-bb36d7ad1d9d.png)



![image](https://user-images.githubusercontent.com/17976289/50315964-d3344f80-04c5-11e9-942a-3dc3f6d081ee.png)
